REditorSupport/languageserver
# Add installation steps of OS-specific dependencies

## What does this PR do ?

Embed three changes, with the dependency commands being the most important one :

1. Fix typos in README
1. Detail how to install requirements before installing `languageserver` package
1. Add table of content in README

## Motivation

Second commit is here to help troubleshooting installation problems on common distros

Third commit eases readability of the repo main page

## Testing Guidelines

All tests were run inside docker images, as summarized in https://github.com/badouralix/dockerfiles/tree/master/r-languageserver

Besides missing compiling tools, namely three errors occurred with fairly easy fixes

All of them showed up when I ran the following command ( see `curl`, `openssl` and `xml2` dependencies )

```bash
$ Rscript -e 'install.packages("languageserver", repos="https://cloud.r-project.org/")'
Installing package into '/usr/local/lib/R/site-library'
(as 'lib' is unspecified)
also installing the dependencies 'prettyunits', 'sys', 'pkgbuild', 'askpass', 'ps', 'lazyeval', 'remotes', 'ellipsis', 'evaluate', 'pkgload', 'praise', 'curl', 'mime', 'openssl', 'highr', 'markdown', 'stringr', 'yaml', 'Rcpp', 'fansi', 'utf8', 'vctrs', 'glue', 'R.methodsS3', 'R.oo', 'R.utils', 'lifecycle', 'pkgconfig', 'processx', 'assertthat', 'crayon', 'rprojroot', 'rex', 'cyclocomp', 'testthat', 'digest', 'rstudioapi', 'httr', 'knitr', 'htmltools', 'pillar', 'base64enc', 'backports', 'cli', 'magrittr', 'purrr', 'R.cache', 'rematch2', 'rlang', 'tibble', 'withr', 'xfun', 'callr', 'collections', 'desc', 'fs', 'jsonlite', 'lintr', 'R6', 'repr', 'stringi', 'styler', 'xml2', 'xmlparsedata'
```

### libcurl

```text
* installing *source* package ‘curl’ ...
** package ‘curl’ successfully unpacked and MD5 sums checked
** using staged installation
Using PKG_CFLAGS=
Using PKG_LIBS=-lcurl
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because libcurl was not found. Try installing:
 * deb: libcurl4-openssl-dev (Debian, Ubuntu, etc)
 * rpm: libcurl-devel (Fedora, CentOS, RHEL)
 * csw: libcurl_dev (Solaris)
If libcurl is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a libcurl.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------------------------------------------------
ERROR: configuration failed for package ‘curl’
* removing ‘/usr/lib/R/library/curl’
```

### libssl

```text
* installing *source* package 'openssl' ...
** package 'openssl' successfully unpacked and MD5 sums checked
Using PKG_CFLAGS=
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because openssl was not found. Try installing:
 * deb: libssl-dev (Debian, Ubuntu, etc)
 * rpm: openssl-devel (Fedora, CentOS, RHEL)
 * csw: libssl_dev (Solaris)
 * brew: openssl@1.1 (Mac OSX)
If openssl is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a openssl.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------------------------------------------------
ERROR: configuration failed for package 'openssl'
* removing '/usr/local/lib/R/site-library/openssl'
```

### libxml2

```text
* installing *source* package 'xml2' ...
** package 'xml2' successfully unpacked and MD5 sums checked
Using PKG_CFLAGS=
Using PKG_LIBS=-lxml2
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because libxml-2.0 was not found. Try installing:
 * deb: libxml2-dev (Debian, Ubuntu, etc)
 * rpm: libxml2-devel (Fedora, CentOS, RHEL)
 * csw: libxml2_dev (Solaris)
If libxml-2.0 is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a libxml-2.0.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------------------------------------------------
ERROR: configuration failed for package 'xml2'
* removing '/usr/local/lib/R/site-library/xml2'
```
